### ISSUE TITLE
Proposal: Style/RegexpLiteral AllowInnerBackslashes

### ISSUE THREAD

**Issue by schwern** [rubocop-hq/rubocop#7568]
`Style/RegexpLiteral` has `AllowInnerSlashes`, but it does not take into account backslashes. With `EnforcedStyle: 'mixed'` and `AllowInnerSlashes: false` I find Rubocop's corrections more difficult to read.

```Ruby
Style/RegexpLiteral:
  EnforcedStyle: 'mixed'
  AllowInnerSlashes: false

# bad?
%r{\A([a-z0-9\-_]+\.)+[a-z]+\z}im,

# good?
/\A([a-z0-9\-_]+\.)+[a-z]+\z/im,

# bad?
%r{\A\d{5}(-\d{4})?\z},

# good?
/\A\d{5}(-\d{4})?\z/,
```

I propose `AllowInnerBackslashes` to work in a manner similar to `AllowInnerSlashes` but for `\`.

```Ruby
Style/RegexpLiteral:
  EnforcedStyle: 'mixed'
  AllowInnerSlashes: false
  AllowInnerBackslashes: false

# good
%r{\A([a-z0-9\-_]+\.)+[a-z]+\z}im,

# bad
/\A([a-z0-9\-_]+\.)+[a-z]+\z/im,

# good
%r{\A\d{5}(-\d{4})?\z},

# bad
/\A\d{5}(-\d{4})?\z/,
```

If folks think that's a good idea, I'm happy to write up a PR.

**Comment by buehmann** [comment]
What would be the meaning of `AllowInnerBackslashes`? If the regexp contains one or more backslashes, the cop will always recommend `%r` instead of `/` as delimiters?

That would not usually reduce the number of backslashes in the regexp, would it? (that for me is the point of `AllowInnerSlashes: false`: to avoid backslashes)

**Comment by schwern** [comment]
@buehmann Ideally we'd have "allow", "disallow", and "don't care" with "don't care" (ie. do not modify the `EnforcedStyle`) being the default. That would be `true`, `false`, and... `nil` if we don't mind trinary logic, and a string if you do. One would never actually write `AllowInnerBackslashes: nil`, it's the default.

I don't see `AllowInnerBackslashes` to reduce the number of backslashes, but to reduce the visual confusion of mixing forward and backslashes. Especially in shorter regexes.
